src/mame/drivers/plygonet.cpp

```diff
@@ -824,11 +824,18 @@ TILE_GET_INFO_MEMBER(polygonet_state::ttl_get_tile_info)
 
 TILE_GET_INFO_MEMBER(polygonet_state::roz_get_tile_info)
 {
-	const auto roz_vram = util::big_endian_cast<const u16>(m_roz_vram.target());
-	const int code = roz_vram[tile_index] & 0x7ff;
-	const int attr = (roz_vram[tile_index] >> 12) + 16; // ROZ base palette is palette index 16 onward
+	if (tile_index < 0x800)
+	{
+		const auto roz_vram = util::big_endian_cast<const u16>(m_roz_vram.target());
+		const int code = roz_vram[tile_index] & 0x7ff;
+		const int attr = (roz_vram[tile_index] >> 12) + 16; // ROZ base palette is palette index 16 onward
+		tileinfo.set(0, code, attr, 0);
+	}
+	else
+	{
+		tileinfo.set(0, 0, 0, 0);
+	}
```

Comment:
This seems rather hacky – tilemap is twice as big but half of it is all tile zero?  Does that really sound like a realistic implementation?

Reply:
As explained on IRC and in the shoutbox, I think I understand why this works, why it can't really be fixed in the near term, and why I'm okay with this change going in as long as it's denoted in code as a work-around:
- The K053936 most likely independently supports wrap vs. clamp behavior on either axis, rather than both or none.
- MAME's tilemap system does not support wrap vs. no-wrap behavior on independent axes.
- MAME's tilemap system is an unrealistic representation of tilemaps, as it's wrap vs. nothing, whereas an actual tilemap most likely outputs *something* outside of its normal bounds, just not what you might want.
- Polygonet and Poly-Net have a fixed layer mixing configuration with the PSAC2 at the back. Most likely scenario is that the PSAC2 is always expected to be outputing something.

Reply:
Follow-up: After getting the low-down from OG on how the PSAC2 actually works, this change is in fact completely reasonable and should go in as-is, without the need for additional commenting.

The long and short of it is that the K053936 implementation only bears a passing resemblance to reality, and in reality it would be easily doable for hardware to trigger things like priority changes or even blanking based on bits in the tile index.

Reply:
My suspicion is that the tilemap generator might be generating a 64x64 tilemap but without the RAM to support it, so pulling 'blank' tilemap data for the lower half.

It's possible also that wraparound on y should be turned off only, the game definitely needs wraparound on x.  MAME's tilemap systmem doesn't support this.  As MG says, even after that some real tilemap hardware can sometimes output a configured value, or nothing at all outside of that space.

Any given solution would be a guess, either could be implemented in hardware, this I just feel was the least invasive for lack of other information.

Reply:
No sweat - the pull request against the release branch has been submitted, and this should be able to be brought in once the file re-organization is done.

Reply:
btw @cuavas I've closed this as @MooglyGuy created a new version that should work with release https://github.com/mamedev/mame/pull/9990